pkg/stores/sqlpartition/queryparser/selector.go

```diff
@@ -0,0 +1,965 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file is derived from
+https://github.com/kubernetes/apimachinery/blob/90df4d1d2d40ea9b3a522bec6e3577237358de00/pkg/labels/selector.go
+*/
+
+/**
+Main changes:
+
+1. The upstream `selector.go` file does parsing and applying to the objects being test.
+We only care about the parser, so the selection part is dropped.
+
+2. I dropped label value validation in the parser
+
+3. Multiple values are returned as an array rather than a `k8s.io/utils/sets.String` object
+to avoid having to pull in that dependency as well (and it isn't needed because we convert
+the array into a sql statement. So the set gives us no benefit apart from removing duplicate target values).
+
+4. We added the `QuotedStringToken` constant to distinguish exact matches from substring matches.
+This needed the `SingleQuoteToken` and `DoubleQuoteToken` variants for the lexer.
+
+5. Our filter language ignores case for `in` and `notin`. These must be lower-case in kubectl filter expressions.
+
+6. The `Lexer.Lex` function names the return parameters in its header but has no argument-less
+   return statement, so I dropped the names.
+
+7.  We allow `lt` and `gt` as aliases for `<` and `>`.
+*/
+
+package queryparser
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var (
+	unaryOperators = []string{
+		string(selection.Exists), string(selection.DoesNotExist),
+	}
+	binaryOperators = []string{
+		string(selection.In), string(selection.NotIn),
+		string(selection.Equals), string(selection.DoubleEquals), string(selection.NotEquals),
+		string(selection.GreaterThan), string(selection.LessThan),
+	}
+	validRequirementOperators = append(binaryOperators, unaryOperators...)
+	labelSelectorRegex        *regexp.Regexp
```

Comment:
```suggestion
	labelSelectorRegex        regexp.MustCompile(...)
```

This should work since you're in a `var` block

Reply:
Yes